pipeline-scripts/release.groovy

```diff
@@ -673,7 +673,7 @@ def getReleaseTagArchPriv(from_release_tag) {
         priv = true
     } else if (!nameComponents[2].isNumber()) {
         arch = commonlib.brewArchForGoArch(nameComponents[2])
-        priv = nameComponents[3] == "priv"
+        if (nameComponents.length > 3 && nameComponents[3] == "priv") priv = true
```

Comment:
This was the source of the bug where a normal tag like `4.12.0-ec.2-s390x` would hit a nondescript `java.lang.ArrayIndexOutOfBoundsException` - because it's expecting a nightly tag with at least 4 elements apparently; but we don't sync nightlies as dev-previews anymore and this didn't do the right thing for other tags.